src/main/java/fi/metatavu/metaform/server/metaforms/ReplyController.java

```diff
@@ -313,7 +366,180 @@ public void deleteReply(Reply reply) {
   public List<ReplyField> listReplyFields(Reply reply) {
     return anyReplyFieldDAO.listByReply(reply);
   }
+  
+  /**
+   * Returns table field as valid table data. 
+   * 
+   * If input object is not valid null is returned instead
+   * 
+   * @param columnMap column map
+   * @param value input object
+   * @return table data
+   */
+  @SuppressWarnings("unchecked")
+  private List<Map<String, Object>> getTableValue(Map<String, MetaformTableColumn> columnMap, Object value) {
+    if (value == null) {
+      return Collections.emptyList();
+    }
+    
+    if (!(value instanceof List)) {
+      return null;
+    }
+    
+    List<Object> listValue = (List<Object>) value;
+    if (listValue.isEmpty()) {
+      return Collections.emptyList();
+    }
+    
+    for (Object listItem : listValue) {
+      if (!(listItem instanceof Map)) {
+        return null;
+      }
+      
+      Map<Object, Object> mapItem = (Map<Object, Object>) listItem;
+      for (Object key : mapItem.keySet()) {
+        if (!(key instanceof String)) {
+          return null;
+        }
+        
+        String columnName = (String) key;
+        MetaformTableColumn column = columnMap.get(columnName);
+        
+        if (column == null || !isSupportedTableColumnType(column.getType())) {
+          return null;
+        }
+      }
+    }
+    
+    return (List<Map<String, Object>>) value;
+  }
+  
+  /**
+   * Sets reply field value
+   * 
+   * @param field field
+   * @param reply reply
+   * @param name name 
+   * @param values values
+   * @return updated field
+   */
+  private ReplyField setTableReplyField(MetaformField field, Reply reply, String name, Object value) {
+    Map<String, MetaformTableColumn> columnMap = getTableColumnMap(field);
+    List<Map<String, Object>> tableValue = getTableValue(columnMap, value);
+    if (tableValue == null) {
+      if (logger.isErrorEnabled()) {
+        logger.error(String.format("Invalid value (%s) passes to table field %s in reply %s", String.valueOf(value), name, reply.getId().toString()));
+      }
+      
+      return null;
+    }
+    
+    ReplyField replyField = anyReplyFieldDAO.findByReplyAndName(reply, name);
+    if (replyField != null && !(replyField instanceof TableReplyField)) {
+      deleteField(replyField);
+      replyField = null;
+    }
+    
+    TableReplyField tableReplyField = (TableReplyField) replyField;
+    if (tableReplyField == null) {
+      tableReplyField = tableReplyFieldDAO.create(UUID.randomUUID(), reply, name);
+    } else {
+      deleteTableReplyFieldRows(tableReplyField);
+    }
+    
+    for (Map<String, Object> rowValue : tableValue) {
+      createTableReplyFieldRowValue(tableReplyField, columnMap, rowValue);
+    }
+
+    return tableReplyField;
+  }
 
+  private Map<String, MetaformTableColumn> getTableColumnMap(MetaformField field) {
+    Map<String, MetaformTableColumn> columnMap = field.getColumns().stream()
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Immediately return this expression instead of assigning it to the temporary variable "columnMap". ([squid:S1488](https://sonarcloud.io/organizations/metatavu-oy/rules#rule_key=squid%3AS1488))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=fi.metatavu.metaform-api%3Ametaform-api&pullRequest=60&issues=AWY4b1pzvOWp7Yd6rqjk&open=AWY4b1pzvOWp7Yd6rqjk)

[ISSUE_KEY]: AWY4b1pzvOWp7Yd6rqjk